includes/Asset_Loader.php

```diff
@@ -32,15 +32,23 @@ class Asset_Loader {
 	 *
 	 * @since 0.1.0
 	 *
-	 * @param string $handle The handle for the script.
-	 * @param string $file_name The script file name.
+	 * @param string                    $handle     The handle for the script.
+	 * @param string                    $file_name  The script file name.
+	 * @param array<string, mixed>|null $asset_data Optional asset metadata (dependencies and version).
 	 */
-	public static function enqueue_script( string $handle, string $file_name ): void {
+	public static function enqueue_script( string $handle, string $file_name, ?array $asset_data = null ): void {
 		$script_path       = AI_EXPERIMENTS_DIR . 'build/' . $file_name . '.js';
 		$script_url        = AI_EXPERIMENTS_PLUGIN_URL . 'build/' . $file_name . '.js';
-		$script_asset_path = $script_path . '.asset.php';
+		$script_asset_path = substr( $script_path, 0, -3 ) . '.asset.php';
```

Comment:
Ah, this was my mistake. I'd suggest instead of this `substr` call, we just remove the appending of `.js` to `$script_path`, as that's not needed (and was my mistake)
```suggestion
		$script_path       = AI_EXPERIMENTS_DIR . 'build/' . $file_name;
		$script_url        = AI_EXPERIMENTS_PLUGIN_URL . 'build/' . $file_name . '.js';
		$script_asset_path = $script_path . '.asset.php';
```

Reply:
How that? If we remove appending `.js`, that path won't be correct, as it'll be missing the extension.

Reply:
I was thinking we don't ever actually use that path other than to get the `asset.php` file but I guess we do use it as a fallback in the `filemtime` call which would fail in that case